### PR TITLE
RELATED: RAIL-4106 menu button render mode

### PR DESCRIPTION
**Pull request by MilanLempera**
- add delete button and delete dialog
 - update menu items visibility in edit mode

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7994/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9807/ : FAILURE in 5m 36s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8000/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9818/ : SUCCESS in 16m 26s

File: libs/sdk-ui-dashboard/.dependency-cruiser.js [hunk 1]
```diff
@@ -36,6 +36,8 @@ options = {
                 "src/presentation/scheduledEmail",
                 "src/presentation/shareDialog",
                 "src/presentation/saveAs",
+                "src/presentation/deleteDialog",
+                "src/presentation/saveAs",
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/.dependency-cruiser.js:40]: `src/presentation/saveAs` is mentioned twice now, please remove one.

File: libs/sdk-ui-dashboard/src/presentation/saveAs/DefaultSaveAsDialog/index.tsx [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+} from "../../../model";
+
+/**
+ * @internal
+ */
+export function useSaveAsDialogProps(): ISaveAsDialogProps {
+    const { addSuccess, addError } = useToastMessage();
+
+    const dispatch = useDashboardDispatch();
+    const closeSaveAsDialog = () => dispatch(uiActions.closeSaveAsDialog());
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/saveAs/DefaultSaveAsDialog/index.tsx:21]: This is created on every invocation but in the onSaveAsError only uses the first version. Should this also be a useCallback?

File: libs/sdk-ui-dashboard/.dependency-cruiser.js (hunk 1 above)
Comment (MilanLempera) [reply libs/sdk-ui-dashboard/.dependency-cruiser.js:39]: fixed

File: libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts [hunk 3]
```diff
@@ -184,7 +184,7 @@ export const selectEnableKPIDashboardSchedule = createSelector(selectConfig, (st
  * @public
  */
 export const selectEnableKPIDashboardScheduleRecipients = createSelector(selectConfig, (state) => {
-    return state.settings?.enableKPIDashboardScheduleRecipients;
+    return state.settings?.enableKPIDashboardScheduleRecipients ?? false;
```
[the remark this answers is not in the document]
Comment (MilanLempera) [reply libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts:187]: I prefer output type boolean, before boolean | undefined - it breaks types in create selector function

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2286/ : SUCCESS in 2m 37s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2578/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1398/ : SUCCESS in 19m 35s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9820/ : SUCCESS in 17m 23s

**Pull request merged**